src/DNS/Adapter/Native.php

```
@@ -94,4 +190,98 @@ public function getName(): string
+        $chunkSize = strlen($chunk);
+
+        if ($currentBufferSize + $chunkSize > $this->maxTcpBufferSize) {
+            printf("TCP buffer size limit exceeded for client %d\n", $clientId);
+            $this->closeTcpClient($client);
+            return;
+        }
+
+        $this->tcpBuffers[$clientId] = ($this->tcpBuffers[$clientId] ?? '') . $chunk;
+
+        while (strlen($this->tcpBuffers[$clientId]) >= 2) {
+            $length = unpack('nlen', substr($this->tcpBuffers[$clientId], 0, 2));
```

PHPMD (2.15.0)

Avoid unused local variables such as '$length'. (undefined)

(UnusedLocalVariable)